File: 36-valid-sudoku/36-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        
        set<string> s;
        
        for(int i=0; i<9; i++){
            for(int j=0; j<9; j++){
                
                char ch = board[i][j];
                
                if(ch != '.'){
                    
                    string no="";
                    no +=ch;

                    string ith = to_string(i);
                    string jth = to_string(j);


                    if(s.find(no + "_R_" + ith) != s.end() || 
                      s.find(no + "_C_" + jth) != s.end() ||
                      s.find(no + "_B_" + to_string(i/3) + to_string(j/3)) != s.end()) return false;

                    else{
                        s.insert(no + "_R_" + ith);
                        s.insert(no + "_C_" + jth);
                        s.insert(no + "_B_" + to_string(i/3) + to_string(j/3));
                    }
                }
            }
        }
        
        return true;
    }
};

File: 36-valid-sudoku/36-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        
        hashset = set()
        
        for i in range(9):
            for j in range(9):
                
                if board[i][j] == ".":
                    continue
                    
                ith = str(i)
                jth = str(j)
                
                no = ""
                no += board[i][j]
                
                if (no+"r"+ith in hashset or 
                    no+"c"+jth in hashset or 
                    no+"b"+ str(i//3) + str(j//3) in hashset):
                    return False
                
                hashset.add(no+"r"+ith)
                hashset.add(no+"c"+jth)                
                hashset.add(no+"b"+str(i//3)+str(j//3))    
        
        return True